tools/CouchDB_Import_Instruments.php

```diff
@@ -104,23 +104,18 @@ function generateDocumentSQL($instrument, $JSONData)
                         f.CommentID NOT LIKE 'DDE%' 
                         AND s.Active='Y' AND c.Active='Y'";
 
-        if ($JSONData) {
-            // the data is in the flag table, add the data column to the query
+        if ($tablename === "") {
```

Comment:
Why depend on a string being empty or not when you have a more "explicit" boolean telling you what query you should run here? any gain ?

that said I agree with the changes below in the `UpdateCandidateDocs` functions.

Reply:
one is directly derivable from the other, passing them both gives the possibility of them contradicting each other and I don't really find that the boolean is any clearer.. 
if there's no table to join get data from the flag table" seems reasonable to me.

I don't really feel strongly about it, I just don't see much reason to needlessly pass things around where it isn't necessary.